### PR TITLE
feat(headless)!: use next as default analyticsMode

### PR DESCRIPTION
**Pull request by fbeaudoincoveo**
https://coveord.atlassian.net/browse/KIT-3525

@erocheleau I cherry picked the commit you made in https://github.com/coveo/ui-kit/pull/4385 to set analyticsMode: legacy on all Quantic interfaces.

### CODE REVIEW

File: packages/headless/esbuild.mjs [hunk 1]
```diff
@@ -200,7 +200,6 @@ async function buildBrowserConfig(options, outDir) {
       alias({
         'coveo.analytics': resolveEsm('coveo.analytics'),
         pino: resolveBrowser('pino'),
-        '@coveo/pendragon': resolve('./ponyfills', 'magic-cookie-browser.js'),
```
Comment (fbeaudoincoveo) [inline packages/headless/esbuild.mjs:203]: @louis-bompart I presume we no longer need the magic cookie

Comment (fbeaudoincoveo) [reply packages/headless/esbuild.mjs:203]: As discussed in person, we'll keep it after all so I'll revert the cookie-related changes in this PR.

**Review by louis-bompart — APPROVED** [review]

**Review by jpmarceau — COMMENTED** [review]
Curious: why stick to `legacy` in several tests and sample pages? Is the plan to refactor them in a later task?

**Comment by alexprudhomme** [comment]
> Curious: why stick to `legacy` in several tests and sample pages? Is the plan to refactor them in a later task?

The plan is to eventually rewrite all tests in playwright and use the default next mode. It would take more time to readjust the test now then simply put `legacy`.

**Review by alexprudhomme — APPROVED** [review]

**Pull request merged**